fw/t/unit/test_http_parser.c

```diff
@@ -148,14 +153,20 @@ split_and_parse_n(unsigned char *str, int type, size_t len, size_t chunk_size)
 			  len, pos);
 		if (type == FUZZ_REQ)
 			r = tfw_http_parse_req(req, str + pos, chunk_size, &parsed);
+		else if (type == FUZZ_REQ_H2)
+			r = tfw_h2_parse_req(req, str + pos, chunk_size, &parsed);
 		else
 			r = tfw_http_parse_resp(resp, str + pos, chunk_size, &parsed);
 
 		pos += chunk_size;
 		hm->msg.len += parsed;
 
+		BUILD_BUG_ON((int)TFW_POSTPONE != (int)T_POSTPONE);
 		if (r != TFW_POSTPONE)
 			return r;
+
+		if (type == FUZZ_REQ_H2)
+			r = tfw_h2_parse_req_finish(req);
```

Comment:
There is drastic logic error here. `tfw_h2_parse_req` never return `T_OK`, but `T_DROP` or `T_POSTPONE`. Here you unconditionally call `tfw_h2_parse_req_finish` that almost always returns `T_OK`. In fact all your h2 tests below not even fully ran.

Reply:
OK. It seems you are right. I'll try to fix this.